code/mathematical-algorithms/Check_is_square/square_checker.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func square_check(a float64) int {
+
+	var int_root int
+	var float_root float64
+	
+	int_root = int(math.Sqrt(a))
+	float_root = math.Sqrt(a)
```

Comment:
Remove one call to sqrt:

```go
float_root = math.Sqrt(a)
int_root = int(float_root)
```
On the other hand, you don't need the float root, you can simplify this whole function to:

```go
func square_check(a int) bool {
	 var int_root int = int(math.Sqrt(a))
	 return (int_root * int_root) == a
}
```

Reply:
Forgive me for any errors with the code. I am inexperienced with Go.